rfcs/convergence/react-focus-management.md

```diff
@@ -0,0 +1,182 @@
+
+This section is important as the motivation or problem statement is indepenent from the proposed change. Even if this RFC is not accepted this Motivation can be used for alternative solutions.
+
+In the end, please make sure to present a neutral Problem statement, rather than one that motivates a particular solution
+-->
+
+The current `FocusZone` and `FocusTrapZone` are React components that are responsible for keyboarding and focus trap behaviours in v7 and v0. The main difference between the two libraries is that v7 makes these components publicly available while v0 does not.
+
+The React component pattern for focus management is not ideal, since it breaks all React paradigms for component isolation. These components rely on complex DOM operations and event listeners to implement behaviour. The components also expose a class-based interface for focus operations on children elements, which further breaks this paradigm.
+
+The current `FocusZone` components apply focus behaviours on **rendered DOM** as opposed to **visible DOM**. The Teams messaging stream is an example of what can happen if focus lands from rendered items, it can be an issue virtualization and may cause unnecessary scrolling which will:
+
+- Force rerender items that aren't visible
+- Cause unnecessary network requests due to force rerendering
+- Extra narration for screen reader users with unnecessary items not rendered
+
+Nested focusables are a source of issues on the current `FocusZone` component, where nested focusables behave inconsistently or incorrectly. Here are some recent issues related to this
+
+- [microsoft/fluentui#8551](https://github.com/microsoft/fluentui/issues/8551)
+- [microsoft/fluentui#12177](https://github.com/microsoft/fluentui/issues/12177)
+- [microsoft/fluentui#16037](https://github.com/microsoft/fluentui/issues/16037)
+- [microsoft/fluentui#13210](https://github.com/microsoft/fluentui/issues/13210)
+
+## Detailed Design or Proposal
+
+### Ability helpers
+
+<!-- This is the bulk of the RFC. Explain the proposal or design in enough detail for the inteded audience to understand. -->
+
+The `@fluentu/react-focus-management` package will act as a facade wrapper around [ability-helpers](https://github.com/microsoft/ability-helpers) and exposes a set of hooks and utilities for focus management. The [ability-helpers](https://github.com/microsoft/ability-helpers) library is designed to be a lower level utility that manages the state of focusable elements than any rendering framework.
+
+[ability-helpers](https://github.com/microsoft/ability-helpers) manages focus on a lower level than the rendering framework and acts on the **visible** DOM. Since operations are run on visible DOM there is a much clearer path for handling dynamically loaded content that would otherwise require effects outside the rendering lifecycle in React components which breaks component isolation principles.
```

Comment:
nit:
```suggestion
[tabster](https://github.com/microsoft/tabster) manages focus on a lower level than the rendering framework and acts on the **visible** DOM. Since operations are run on visible DOM there is a much clearer path for handling dynamically loaded content that would otherwise require effects outside the rendering lifecycle in React components which breaks component isolation principles.
```